### PR TITLE
disable license check for new formulae

### PR DESCRIPTION
**Pull request by chenrui333**
- [ ] Have you followed the guidelines in our [Contributing](https://github.com/Homebrew/brew/blob/HEAD/CONTRIBUTING.md) document?
- [ ] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/brew/pulls) for the same change?
- [ ] Have you added an explanation of what your changes do and why you'd like us to include them?
- [ ] Have you written new tests for your changes? [Here's an example](https://github.com/Homebrew/brew/blob/HEAD/Library/Homebrew/test/PATH_spec.rb).
- [ ] Have you successfully run `brew style` with your changes locally?
- [ ] Have you successfully run `brew tests` with your changes locally?

-----
relates to 
- https://github.com/Homebrew/homebrew-core/pull/57237
- https://github.com/Homebrew/homebrew-core/pull/57242

cc @Bo98

### CODE REVIEW

File: Library/Homebrew/dev-cmd/audit.rb [hunk 1]
```diff
@@ -333,23 +333,19 @@ def audit_formula_name
     ].freeze
 
     def audit_license
-      if formula.license.present?
-        if @spdx_data["licenses"].any? { |lic| lic["licenseId"] == formula.license }
-          return unless @online
+      if @spdx_data["licenses"].any? { |lic| lic["licenseId"] == formula.license }
```
Comment (Bo98) [inline Library/Homebrew/dev-cmd/audit.rb:336]:
```suggestion
      return if formula.license.blank?
      if @spdx_data["licenses"].any? { |lic| lic["licenseId"] == formula.license }
```

**Comment by SMillerDev** [comment]
Why not just return successfully at the start of the method?

**Comment by Bo98** [comment]
Unreachable code is a style error, unless I misunderstand what you mean.

**Comment by SMillerDev** [comment]
No, I'm just not aware of all the style rules

**Comment by Bo98** [comment]
@MikeMcQuaid Please revert either just before or just after 2.4.3.

**Pull request merged**